include/micm/solver/cuda_lu_decomposition.cuh

```diff
@@ -3,13 +3,27 @@
 // SPDX-License-Identifier: Apache-2.0
 #pragma once
 #include <micm/util/cuda_param.hpp>
-#include <vector>
 namespace micm
 {
   namespace cuda
   {
+    /// This is the host function that will call the CUDA kernel
+    ///   to perform LU decomposition on the device
     std::chrono::nanoseconds DecomposeKernelDriver(
             CudaSparseMatrixParam& sparseMatrix, 
-            CudaSolverParam& solver);
-  }  // namespace cuda
-}  // namespace micm
+            LuDecomposeConst* devptr);
```

Comment:
```suggestion
            LuDecomposeConst& devptr);
```
I think, in most cases, it's best to prefer reference over pointer arguments

Reply:
I just update the pointers to be references.